### ISSUE TITLE
input() already active 500 error at /rotation

### ISSUE THREAD

**Issue by bakert** [PennyDreadfulMTG/Penny-Dreadful-Tools#4668]
Seen a few of these.

Probably move off using fileinput?

https://stackoverflow.com/questions/13799880/what-does-input-already-active-mean-in-python-fileinput-module

```
  Reported on decksite by 158367381427585024 Request Method: GET Path: /rotation/? Cookies: {'__cfduid': 'd9a207d24f32ca79d3d3233b63cc82c9b1522653091', '_ga': 'GA1.2.707512550.1522653104', 'hide_intro': 'True', 'deck_id': '12992', 'session': '.eJxlUF1PwzAM_C9-rlDbNFnYGxuT9gGDoWmCIVRFqddma5qRpBrVtP9OigR7QPKLz3f2nc8gCq0aGO5E7TACVcAQEsoJGxCeZOmAchqnGURQm7LEIj-idabJe17CBiwC7UuTtw5tIzQG8V7YUL5SzcEFmRGtr9LceeH7qR3Pildu7uQLqj3Vz5PVQ9m9FdP2UZJkfOV7c8Dg6gxCSnTutwfzdLtOR5yeDN0SoiWulZuyT1Tcdt086PHrqCy6XPhgkKYkpTwm8Q3LWJaw7DrvM7M443EcgcVdgKq_I1ouT6uFXUzGW7VZFt12rmf31WQk5TxTvUknzTGkeQ_vwsarXRewslV14eAjgp81ue96CoxQWLRwieDf57ywJQabTVvXl2_rd39j.Db2OQg.UwM8LdhuspaFFJMTw36z22HmfW4'} Endpoint: rotation View Args: {} Person: 158367381427585024 Referrer: None Request Data: {} Host: pennydreadfulmagic.com Accept-Encoding: gzip Cf-Ipcountry: CA X-Forwarded-For: 206.47.117.241, 162.158.63.168 Cf-Ray: 410242332a084722-EWR X-Forwarded-Proto: https Cf-Visitor: {"scheme":"https"} Upgrade-Insecure-Requests: 1 User-Agent: Mozilla/5.0 (Windows NT 10.0; Win64; x64) AppleWebKit/537.36 (KHTML, like Gecko) Chrome/65.0.3325.181 Safari/537.36 Accept: text/html,application/xhtml+xml,application/xml;q=0.9,image/webp,image/apng,/;q=0.8 Accept-Language: en-US,en;q=0.9 Cookie: __cfduid=d9a207d24f32ca79d3d3233b63cc82c9b1522653091; _ga=GA1.2.707512550.1522653104; hide_intro=True; deck_id=12992; session=.eJxlUF1PwzAM_C9-rlDbNFnYGxuT9gGDoWmCIVRFqddma5qRpBrVtP9OigR7QPKLz3f2nc8gCq0aGO5E7TACVcAQEsoJGxCeZOmAchqnGURQm7LEIj-idabJe17CBiwC7UuTtw5tIzQG8V7YUL5SzcEFmRGtr9LceeH7qR3Pildu7uQLqj3Vz5PVQ9m9FdP2UZJkfOV7c8Dg6gxCSnTutwfzdLtOR5yeDN0SoiWulZuyT1Tcdt086PHrqCy6XPhgkKYkpTwm8Q3LWJaw7DrvM7M443EcgcVdgKq_I1ouT6uFXUzGW7VZFt12rmf31WQk5TxTvUknzTGkeQ_vwsarXRewslV14eAjgp81ue96CoxQWLRwieDf57ywJQabTVvXl2_rd39j.Db2OQg.UwM8LdhuspaFFJMTw36z22HmfW4 Cf-Connecting-Ip: 206.47.117.241 X-Forwarded-Host: pennydreadfulmagic.com X-Forwarded-Server: pennydreadfulmagic.com Connection: Keep-Alive-------------------------------------------------------------------------------- RuntimeError Stack Trace: File "/home/discord/.local/lib/python3.6/site-packages/flask/app.py", line 1997, in call return self.wsgi_app(environ, start_response) File "/home/discord/.local/lib/python3.6/site-packages/flask/app.py", line 1985, in wsgi_app response = self.handle_exception(e) File "/home/discord/.local/lib/python3.6/site-packages/flask/app.py", line 1982, in wsgi_app response = self.full_dispatch_request() File "/home/discord/.local/lib/python3.6/site-packages/flask/app.py", line 1614, in full_dispatch_request rv = self.handle_user_exception(e) File "/home/discord/.local/lib/python3.6/site-packages/flask/app.py", line 1517, in handle_user_exception reraise(exc_type, exc_value, tb) File "/home/discord/.local/lib/python3.6/site-packages/flask/_compat.py", line 33, in reraise raise value File "/home/discord/.local/lib/python3.6/site-packages/flask/app.py", line 1612, in full_dispatch_request rv = self.dispatch_request() File "/home/discord/.local/lib/python3.6/site-packages/flask/app.py", line 1598, in dispatch_request return self.view_functionsrule.endpoint File "./decksite/cache.py", line 71, in decorated_function response = make_response(f(*args, **kwargs)) File "./decksite/main.py", line 214, in rotation view = Rotation() File "./decksite/views/rotation.py", line 36, in init self.read_rotation_files() File "./decksite/views/rotation.py", line 49, in read_rotation_files for line in fileinput.input(files): File "/usr/lib64/python3.6/fileinput.py", line 93, in input raise RuntimeError("input() already active")
```

**Comment by bakert** [comment]
8 of these in quick succession on July 7 and one today. So it still happens during rotation but rarely.

**Comment by bakert** [comment]
10 more over the last couple of days.

**Comment by bakert** [comment]
5 more recently and the page is no longer live til supplemental.

**Comment by bakert** [comment]
Fixed (?) in 7f73bee5 using FileInput as per https://stackoverflow.com/questions/21443601/runtimeerror-input-already-active-file-loop
